### ISSUE TITLE
Rule Request: Created NSLayoutConstraint from anchor and did nothing with it

### ISSUE THREAD

**Issue by RikkiGibson** [realm/SwiftLint#1545]
I have had UI goofups prolonged by the fact that I created a constraint using an anchor and never did anything with it (such as set `isActive = true` or assign it to a variable). There should probably be a rule that warns when this happens.

**Comment by jpsim** [comment]
Could you please provide some code samples that would and would not trigger this rule?

**Comment by RikkiGibson** [comment]
I see it as very similar to the Android warning when you create a Toast but don't show it. 99% of the time when people create constraints they're interested in activating them, or activating them for certain size classes, or they want them to be assigned to a variable so they can be accessed and manipulated in response to certain events.

```swift
let myLabel = UILabel()
let container = UIView()

// Triggers warning because constraint was created and thrown away without being activated
// (maybe can still be found in myLabel.constraints?)
myLabel.centerYAnchor.constraint(equalTo: container.centerYAnchor)

// Doesn't trigger warning because the constraint was explicitly activated or deactivated
myLabel.centerYAnchor.constraint(equalTo: container.centerYAnchor).isActive = true
myLabel.centerYAnchor.constraint(equalTo: container.centerYAnchor).isActive = false

// Doesn't trigger warning because constraint is being assigned to a variable
// (if deeper forms of analysis are available, could warn if constraint is never activated)
let myConstraint = myLabel.centerYAnchor.constraint(equalTo: container.centerYAnchor)

// If for some reason you just wanted to create this constraint without activating,
// not save it to a variable, and not raise the warning, just assign it to _
_ = myLabel.centerYAnchor.constraint(equalTo: container.centerYAnchor)
```

**Comment by ramjyroo** [comment]
In my case - the following custom rule is working
`custom_rules:
  activate_constraint_rule:
    regex: '\.constraint\({1}[A-Za-z0-9. ,:]+\){1}(?!\.)'
    message: "Expected Constaint to be activated"
    severity: error
`
